Astronomy/Astronomy/Model/MarsPhoto.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  MarsRover.swift
+//  Astronomy
+//
+//  Created by Jesse Ruiz on 12/16/19.
+//  Copyright © 2019 Lambda School. All rights reserved.
+//
+
+import Foundation
+
+@objc class MarsPhotoReference: NSObject {
+    
+    @objc let id: Int
+    @objc let sol: Int
+    @objc let camera: Camera
+    @objc let earthDate: Date
+    @objc let imageURL: URL
+    @objc var dictionary: [String: String]?
+    
+    
+    @objc init(id: Int, sol: Int, camera: Camera, earthDate: Date, imageURL: URL) {
+        self.id = id
+        self.sol = sol
+        self.camera = camera
+        self.earthDate = earthDate
+        self.imageURL = imageURL
+    }
+    
+    @objc convenience init(dictionary: [String: String]) {
```

Comment:
* This is not the right type (dictionary is failing to convert to [String : String])
* If it worked, this is a cycle of calling the same init(dictionary:) to infinity (crash)

It should be something like:

```swift
    @objc convenience init(dictionary: [String: Any]) {
//        self.init(dictionary: dictionary)
		let solNumber = dictionary["sol"] as? Int
		
                // FIXME: don't force unwrap and pass all the data parsed if it exists
		self.init(id: 0, sol: solNumber!, camera: Camera(), earthDate: Date(), imageURL: URL(string: "http://apple.com")!)
    }
```